lib/screens/v2/widget_instance/subway_status.ex

```diff
@@ -0,0 +1,625 @@
+  end
+
+  defp alert_affects_gl_trunk?(%Alert{informed_entities: informed_entities}) do
+    gl_trunk_stops =
+      @route_stop_sequences |> Map.get("Green") |> hd() |> Enum.map(&elem(&1, 0)) |> MapSet.new()
+
+    alert_trunk_stops =
+      informed_entities
+      |> Enum.map(fn e -> Map.get(e, :stop) end)
+      |> Enum.reject(&is_nil/1)
+      |> Enum.uniq()
+      |> Enum.into(MapSet.new())
+      |> MapSet.intersection(gl_trunk_stops)
+
+    MapSet.size(alert_trunk_stops) > 0
+  end
+
+  defp is_multi_route?(alert) do
+    length(alert_routes(alert)) > 1
+  end
+
+  def serialize_green_line(grouped_alerts) do
+    green_line_alerts =
+      ["Green-B", "Green-C", "Green-D", "Green-E"]
+      |> Enum.flat_map(fn route -> Map.get(grouped_alerts, route, []) end)
+      |> Enum.uniq()
+
+    multi_route_alerts = Enum.filter(green_line_alerts, &is_multi_route?/1)
+    single_route_alerts = Enum.reject(green_line_alerts, &is_multi_route?/1)
+    trunk_alerts = Enum.filter(multi_route_alerts, &alert_affects_gl_trunk?/1)
+    alert_count = length(green_line_alerts)
+
+    statuses =
+      single_route_alerts
+      |> Enum.group_by(fn alert -> Enum.at(alert_routes(alert), 0) end)
+      |> Enum.map(fn {route, alerts} -> {route, get_green_line_branch_statuses(alerts)} end)
+      |> group_green_line_branch_statuses()
+
+    case trunk_alerts do
+      [] ->
+        # If there are no alerts for the GL trunk, serialize any alerts on the branches
+        serialize_green_line_branch_alerts(statuses, single_route_alerts, alert_count)
+
+      [trunk_alert] ->
+        # If there is a single alert on the GL trunk, combine it with any alerts on the branches
+        serialize_green_line_trunk_alert(trunk_alert, statuses, alert_count)
+
+      _ ->
+        # If there are multiple alerts on the GL trunk, log it and serialize the count
+        _ =
+          Logger.info(
+            "[subway_status_multiple_alerts] route=#{"Green-Trunk"} count=#{alert_count}"
+          )
+
+        %{
+          type: :trunk,
+          route: serialize_route_pill("Green"),
+          status: "#{alert_count} alerts",
+          location: ["mbta.com/alerts/subway", "mbta.com/alerts/subway"]
+        }
+    end
+  end
+
+  defp serialize_green_line_trunk_alert(trunk_alert, statuses, alert_count) do
+    case statuses do
+      [] ->
+        # One GL trunk alert and no branch alerts
+        Map.merge(
+          %{type: :trunk, route: serialize_route_pill("Green")},
+          serialize_alert(trunk_alert, "Green")
+        )
+
+      [status] ->
+        # One GL trunk alert and one GL branch alert
+        trunk_status = get_green_line_branch_status(trunk_alert)
+        %{type: :branch, statuses: [[nil, trunk_status], status]}
+
+      _ ->
+        # One GL trunk alert and 2+ GL branch alerts
+        _ =
+          Logger.info(
+            "[subway_status_multiple_alerts] route=#{"Green-Trunk"} count=#{alert_count}"
```

Comment:
Should `route` in this log list out the branches/say something other than "Green-Trunk"? The multiple alerts are on the branch(es).